### ISSUE TITLE
pip version is messed up

### ISSUE THREAD

**Issue by cornfeedhobo** [ansible/ansible-lint#4415]
##### Summary

```plain
$ pip index versions ansible-lint
WARNING: pip index is currently an experimental command. It may be removed/changed in a future release without prior warning.
ansible-lint (6.22.2)
Available versions: 6.22.2, 6.22.1, 6.22.0, 6.21.1, 6.21.0, 6.20.3, 6.20.2, 6.20.1, 6.20.0, 6.19.0, 6.18.0, 6.17.2, 6.17.1, 6.17.0, 6.16.2, 6.16.1, 6.16.0, 6.15.0, 6.14.6, 6.14.4, 6.14.3, 6.14.2, 6.14.1, 6.14.0, 6.13.1, 6.13.0, 6.12.2, 6.12.1, 6.12.0, 6.11.0, 6.10.2, 6.10.1, 6.10.0, 6.9.1, 6.9.0, 6.8.7, 6.8.6, 6.8.5, 6.8.4, 6.8.3, 6.8.2, 6.8.1, 6.8.0, 6.7.0, 6.6.1, 6.6.0, 6.5.2, 6.5.1, 6.5.0, 6.4.0, 6.3.0, 6.2.2, 6.2.1, 6.2.0, 6.1.0, 6.0.2, 6.0.1, 6.0.0, 5.4.0, 5.3.2, 5.3.1, 5.3.0, 5.2.1, 5.2.0, 5.1.3, 5.1.2, 5.1.1, 5.0.12, 5.0.11, 5.0.10, 5.0.9, 5.0.8, 5.0.7, 5.0.6, 5.0.5, 5.0.4, 5.0.3, 5.0.2, 5.0.1, 5.0.0, 4.3.7, 4.3.6, 4.3.5, 4.3.4, 4.3.3, 4.3.2, 4.3.1, 4.3.0, 4.2.0, 4.1.0, 4.0.1, 4.0.0, 3.5.1, 3.5.0, 3.4.23, 3.4.22, 3.4.21, 3.4.20, 3.4.19, 3.4.18, 3.4.17, 3.4.16, 3.4.15, 3.4.13, 3.4.12, 3.4.11, 3.4.10, 3.4.9, 3.4.8, 3.4.7, 3.4.6, 3.4.5, 3.4.4, 3.4.3, 3.4.1, 3.4.0, 3.3.3, 3.3.2, 3.3.0, 3.2.5, 3.2.4, 3.2.3, 3.2.2, 3.2.1, 3.2.0, 3.1.3, 3.1.2, 3.1.1, 3.1.0, 3.0.1, 3.0.0, 2.7.1, 2.7.0, 2.6.2, 2.6.1, 2.5.0, 2.4.2, 2.4.1, 2.4.0, 2.3.9, 2.3.8, 2.3.6, 2.3.5, 2.3.3, 2.3.2, 2.3.1, 2.3.0, 2.2.0, 2.1.3, 2.1.0, 2.0.3, 2.0.1, 1.0.4, 1.0.3, 1.0.2, 1.0.1, 1.0.0
  INSTALLED: 6.22.2
  LATEST:    6.22.2
```

```plain
A new release of ansible-lint is available: 6.22.2 → 24.10.0
```

The website shows the `> 24.0.0` releases, but they aren't available through pip.

##### Issue Type

- Bug Report

**Comment by konstruktoid** [comment]
https://pypi.org/project/ansible-lint/ shows `ansible-lint 24.10.0`

```sh
$ python3 -m pip install -U ansible-lint
Requirement already satisfied: ansible-lint in .../venvs/ansible/lib/python3.12/site-packages (24.10.0)
[...]
$ ansible-lint --version
ansible-lint 24.10.0 using ansible-core:2.17.6 ansible-compat:24.10.0 ruamel-yaml:0.18.6 ruamel-yaml-clib:0.2.8
```

**Comment by cornfeedhobo** [comment]
@konstruktoid hmm, only difference is that I'm testing with python3.9 and you are using python3.12... :thinking:

**Comment by konstruktoid** [comment]
https://github.com/ansible/ansible-lint/blob/main/pyproject.toml#L40 
`requires-python = ">=3.10"`

**Comment by cornfeedhobo** [comment]
@konstruktoid Yeah, but 3.10 has the same problem - I don't think this is related to `pyproject.toml`

**Comment by konstruktoid** [comment]
which distro and release are you using?

**Comment by cornfeedhobo** [comment]
opensuse 15.6, python installed using pyenv, dependencies managed with pipenv.

**Comment by konstruktoid** [comment]
I'm sorry, but can you provide instructions on how you performed the Python installation?

**Comment by Qalthos** [comment]
> Edit:
> 
>  * testing with 3.10.x has the same issue

`ansible-lint>=24` is absolutely available for Python >= 3.10:

```
% python3.10 -m pip index versions ansible-lint 
WARNING: pip index is currently an experimental command. It may be removed/changed in a future release without prior warning.
ansible-lint (24.10.0)
Available versions: 24.10.0, 24.9.2, 24.9.1, 24.9.0, 24.7.0, 24.6.1, 24.6.0, 24.5.0, 24.2.3, 24.2.2, 24.2.1, 24.2.0, [...]
```

My _guess_ is that your pyenv shims aren't set properly, but this ultimately looks like an issue with your system. Make sure you are using a supported version of Python and try again.
